File: android-app/app 2/src/main/java/com/bandite/sonicwalkscape/utils/DebugLogger.kt
package com.bandite.sonicwalkscape.utils

import android.util.Log
import java.text.SimpleDateFormat
import java.util.Date
import java.util.Locale

/**
 * Categorized debug logger with timestamps
 */
object DebugLogger {

    private const val TAG = "SonicWalkscape"
    private val dateFormat = SimpleDateFormat("HH:mm:ss.SSS", Locale.US)

    enum class Category(val emoji: String) {
        GPS("📍"),
        AUDIO("🔊"),
        NETWORK("🌐"),
        DOWNLOAD("📥"),
        WARNING("⚠️"),
        ERROR("❌"),
        SUCCESS("✅"),
        INFO("ℹ️")
    }

    private fun timestamp(): String = dateFormat.format(Date())

    private fun log(category: Category, message: String) {
        val formattedMessage = "[${timestamp()}] ${category.emoji} $message"
        when (category) {
            Category.ERROR -> Log.e(TAG, formattedMessage)
            Category.WARNING -> Log.w(TAG, formattedMessage)
            else -> Log.d(TAG, formattedMessage)
        }
    }

    fun gps(message: String) = log(Category.GPS, message)

    fun audio(message: String) = log(Category.AUDIO, message)

    fun network(message: String) = log(Category.NETWORK, message)

    fun download(message: String) = log(Category.DOWNLOAD, message)

    fun warning(message: String) = log(Category.WARNING, message)

    fun error(message: String) = log(Category.ERROR, message)

    fun error(message: String, throwable: Throwable) {
        log(Category.ERROR, "$message: ${throwable.message}")
        Log.e(TAG, message, throwable)
    }

    fun success(message: String) = log(Category.SUCCESS, message)

    fun info(message: String) = log(Category.INFO, message)
}

File: mobile-app/ios/SonicWalkscape/SonicWalkscape/Utilities/DebugLogger.swift
//
//  DebugLogger.swift
//  SonicWalkscape
//
//  Enhanced logging utility for debugging GPS and audio behavior
//

import Foundation

enum DebugLogger {
    enum Category: String {
        case gps = "📍 GPS"
        case audio = "🎵 Audio"
        case network = "🌐 Network"
        case download = "⬇️ Download"
        case warning = "⚠️ Warning"
        case error = "❌ Error"
        case success = "✅ Success"
        case info = "ℹ️ Info"
    }

    private static let dateFormatter: DateFormatter = {
        let formatter = DateFormatter()
        formatter.dateFormat = "HH:mm:ss.SSS"
        return formatter
    }()

    static func log(_ message: String, category: Category = .info) {
        let timestamp = dateFormatter.string(from: Date())
        print("[\(timestamp)] \(category.rawValue) \(message)")
    }

    static func gps(_ message: String) {
        log(message, category: .gps)
    }

    static func audio(_ message: String) {
        log(message, category: .audio)
    }

    static func network(_ message: String) {
        log(message, category: .network)
    }

    static func download(_ message: String) {
        log(message, category: .download)
    }

    static func warning(_ message: String) {
        log(message, category: .warning)
    }

    static func error(_ message: String) {
        log(message, category: .error)
    }

    static func success(_ message: String) {
        log(message, category: .success)
    }
}
